### PR TITLE
Cancel commit if analysis fails

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/analysis/AnalysisErrorCallback.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.analysis;
+
+@FunctionalInterface
+public interface AnalysisErrorCallback {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/AnalysisErrorCallback.java]: MAJOR Drop this interface in favor of "java.util.function.Consumer<Exception>". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1711)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarlint/intellij/trigger/SonarLintCheckinHandler.java [hunk 2]
```diff
@@ -89,19 +90,27 @@ public ReturnResult beforeCheckin(@Nullable CommitExecutor executor, PairConsume
     SonarLintSubmitter submitter = SonarLintUtils.get(project, SonarLintSubmitter.class);
     // this will block EDT (modal)
     try {
-      submitter.submitFilesModal(affectedFiles, TriggerType.CHECK_IN);
+      AtomicBoolean error = new AtomicBoolean(false);
```
Comment (janos-ss) [inline src/main/java/org/sonarlint/intellij/trigger/SonarLintCheckinHandler.java]: `volatile boolean` would be enough

Comment (dbmeneses) [inline src/main/java/org/sonarlint/intellij/trigger/SonarLintCheckinHandler.java]: the goal is just to have a mutable boolean that is effectively final, not to be thread safe.

**Pull request merged**